pkgdb2/lib/__init__.py

```diff
@@ -608,6 +608,12 @@ def update_pkg_status(session, pkg_name, pkg_branch, status, user,
                 acl.status = 'Obsolete'
                 session.add(acl)
             session.flush()
+            # If the package is retired everywhere, stop monitoring it
+            pkg = pkglisting.package
+            if pkg.retired_everywhere:
+                pkg.monitor = False
+                session.add(pkg)
```

Comment:
I *think* that happens automatically since it should already be associated with the session from the original query.